test/framework/vmoperator/vmoperator.go

```diff
@@ -361,6 +362,60 @@ func ReconcileDependencies(ctx context.Context, c client.Client, dependenciesCon
 		return retryError
 	}
 
+	// Create the supervisor service in kube-system for the servicediscovery controller to discover and set an IP address
```

Comment:
Let's create an issue to take a look at the non-IP code path in servicediscovery controller that simply doesn't work today (probably we should just return an error in that case, but TBD)

Reply:
I think the non-IP code is valid, but depends on how you setup k8s.

Reply:
I think it's not. According to the errors we're getting IP is mandatory. So this code path can not work: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/393a16983d02d5a2b254e4182002f70329f9dd8f/controllers/vmware/servicediscovery_controller.go#L394

Reply:
I'm not saying the code path to read kube-public can never work, just that if we don't get an IP there)

Reply:
Opened an issue: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/3409